lib/better_html/tree/attribute.rb

```diff
@@ -1,26 +1,30 @@
+require 'ast'
+
 module BetterHtml
   module Tree
     class Attribute
```

Comment:
Is there a reason `Attribute` does not inherit from `::AST::Node` ? It seems there a ton of `is_a?` in this code, which is usually a bad sign. Why are not all node classes descendants of `::AST::Node`?

Reply:
nodes can have some children that are simple strings and other children that are themselves nodes, for example an attribute value with ERB embedded would look like
```ruby
s(: attribute_name, "foo", s(:erb, s(:indicator, "="), nil, s(:code, " value "), nil), "bar")
```
I could make these string parts into nodes if that is typically how it's done, but I thought it may be overkill.

Reply:
I means one extra class to remove an ever increasing amount of branches based on the class. Maybe it's not worth it at the moment, but consider that for the reader it is not obvious what the `else` branch means. Here it's a string so it's not too bad, but in general it is hard to see what it could be as opposed to a closed number of classes that respond to a certain method.